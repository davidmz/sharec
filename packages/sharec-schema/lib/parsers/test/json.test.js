const { fixtures } = require('testUtils')
const { fromJSON, toJSON } = require('./json')

describe('parsers > json', () => {
  const defaultFxt = fixtures('default/json/00-base')
  const defaultCommentsFxt = fixtures('default/json/03-comments')

  const currentMap = new Map().set('foo', 'bar').set('bar', 'baz')
  const defaultFxtMap = new Map(currentMap).set('baz', currentMap)

  describe('fromJSON', () => {
    it('should create Map from given JSON', () => {
      expect(fromJSON(defaultFxt.current)).toEqual(defaultFxtMap)
    })

    it('should create Map from JSON which includes comments', () => {
      expect(fromJSON(defaultCommentsFxt.current)).toEqual(defaultFxtMap)
    })
  })

  describe('toJSON', () => {
    it('should create JSON string from Map', () => {
      expect(toJSON(defaultFxtMap)).toMatchSnapshot()
    })
  })
})
