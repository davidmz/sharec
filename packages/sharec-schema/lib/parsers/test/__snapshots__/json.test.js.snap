// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parsers > json toJSON should create JSON string from Map 1`] = `
"{
  \\"foo\\": \\"bar\\",
  \\"bar\\": \\"baz\\",
  \\"baz\\": {
    \\"foo\\": \\"bar\\",
    \\"bar\\": \\"baz\\"
  }
}"
`;
